src/ui/features/referral-program/shared/useWalletsMeta.ts

```diff
@@ -0,0 +1,21 @@
+import { useQuery } from '@tanstack/react-query';
+import { ZerionAPI } from 'src/modules/zerion-api/zerion-api.client';
+
+export function useWalletsMeta({
+  addresses,
+  enabled = true,
+}: {
+  addresses: string[];
+  enabled?: boolean;
+}) {
+  return useQuery({
+    enabled: enabled && addresses.length > 0,
+    queryKey: ['zpi/getWalletsMeta', addresses],
+    queryFn: async () => {
+      const response = await ZerionAPI.getWalletsMeta({
```

Comment:
This can lead to a problem when there is more then 10 wallets (400 error)
We need to use `getWalletsMetaByChunks`

Reply:
updated ✅